apps/dapp/src/components/Layouts/V2Layout/index.tsx

```diff
@@ -1,21 +1,96 @@
+import { FunctionComponent, SVGProps, useState } from 'react';
+import * as breakpoints from 'styles/breakpoints';
 import { useMediaQuery } from 'react-responsive';
-import { Outlet } from 'react-router-dom';
+import { Outlet, useLocation } from 'react-router-dom';
 
 import styled from 'styled-components';
 import { queryMinTablet, tabletAndAbove } from 'styles/breakpoints';
 import Footer from './Footer';
 import LeftNav from './Nav/LeftNav';
 import MobileNav from './Nav/MobileNav';
 
+import Dashboard from 'assets/icons/dashboard.svg?react';
+import CurrencyExchange from 'assets/icons/currency_exchange.svg?react';
+import Payments from 'assets/icons/payments.svg?react';
+import Candle from 'assets/icons/candle.svg?react';
+import Restore from 'assets/icons/restore.svg?react';
+
+export type MenuNavItem = {
+  label: string;
+  linkTo: string;
+  Logo: FunctionComponent<SVGProps<SVGSVGElement> & { title?: string | undefined }>;
+  selected: boolean;
+};
+
+export type MenuNavItems = Array<MenuNavItem>;
+
+enum V2DashboardLocPaths {
+  Trade = '/v2dapp/trade',
+  Trv = '/v2dapp/dashboard/treasuryreservesvault',
+  Borrow = '/v2dapp/borrow',
+  Ohmage = '/v2dapp/ohmage',
+  Legacy = '/v2dapp/legacy',
+}
+
 const V2Layout = () => {
   const isTabletOrAbove = useMediaQuery({
     query: queryMinTablet,
   });
+  const loc = useLocation();
+  const [menuNavItems, setMenuNavItems] = useState<Array<MenuNavItem>>([
+    {
+      label: 'Trade',
+      linkTo: V2DashboardLocPaths.Trade,
+      Logo: CurrencyExchange,
+      selected: V2DashboardLocPaths.Trade === loc.pathname,
+    },
+    {
+      label: 'Dashboard',
+      linkTo: V2DashboardLocPaths.Trv,
+      Logo: Dashboard,
+      selected: V2DashboardLocPaths.Trv === loc.pathname,
+    },
+    // TODO: Hidden until launch
+    // {
+    //   label: 'Borrow',
+    //   linkTo: V2DashboardLocPaths.Borrow,
+    //   Logo: Payments,
+    //   selected: V2DashboardLocPaths.Borrow === loc.pathname,
+    // },
+    {
+      label: 'Ohmage',
+      linkTo: V2DashboardLocPaths.Ohmage,
+      Logo: Candle,
+      selected: V2DashboardLocPaths.Ohmage === loc.pathname,
+    },
+    {
+      label: 'Legacy',
+      linkTo: V2DashboardLocPaths.Legacy,
+      Logo: Restore,
+      selected: V2DashboardLocPaths.Legacy === loc.pathname,
+    },
+  ]);
+
+  const onSelectMenuNavItems = async (mi: MenuNavItem) => {
+    await setMenuNavItems((prevState) => {
+      const newState = prevState.map((m) => {
+        if (m == mi) {
```

Comment:
Nit: Not a big deal, certainly not a blocker, but would prefer to name these in a more descriptive way such as "previouslySelectedMenuItem" and "selectedMenuItem".